include/fuzzuf/algorithms/afl_symcc/fuzzer.hpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * fuzzuf
+ * Copyright (C) 2021 Ricerca Security
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+#ifndef FUZZUF_INCLUDE_ALGORITHM_AFL_SYMCC_FUZZER_HPP
+#define FUZZUF_INCLUDE_ALGORITHM_AFL_SYMCC_FUZZER_HPP
+
+#include "fuzzuf/algorithms/afl/afl_state.hpp"
+#include "fuzzuf/executor/afl_symcc_executor_interface.hpp"
+#include "fuzzuf/fuzzer/fuzzer.hpp"
+#include "fuzzuf/hierarflow/hierarflow_intermediates.hpp"
+#include "fuzzuf/hierarflow/hierarflow_node.hpp"
+#include "fuzzuf/hierarflow/hierarflow_routine.hpp"
+#include "fuzzuf/utils/common.hpp"
+#include "fuzzuf/utils/map_file.hpp"
+#include "fuzzuf/utils/sha1.hpp"
+#include "fuzzuf/utils/vfs/read_once.hpp"
+#include <array>
+#include <memory>
+#include <unordered_set>
+#include <vector>
+
+namespace fuzzuf::algorithm::afl_symcc {
+
+/**
+ * @class AFLFuzzerTemplate
+ * @brief Wrapper to afl::AFLFuzzerTemplate. This class append two member
+ * functions required to run SymCC.
+ * @tparam State AFL state type. The type is passed to afl::AFLFuzzerTemplate
+ * straightforward.
+ */
+template <typename State>
+struct AFLFuzzerTemplate : public afl::AFLFuzzerTemplate<State> {
+  /**
+   * All constructor arguments are forwarded to afl::AFLFuzzerTemplate
+   */
+  using afl::AFLFuzzerTemplate<State>::AFLFuzzerTemplate;
+
+  /**
+   * Add the input value to case_queue.
+   * @param buf pointer to the input value
+   * @param len length of the input value
+   * @return inserted value in Testcase
+   */
+  auto AddToQueue(const unsigned char *buf, std::uint32_t len) const {
+    const auto fn = Util::StrPrintf(
+        "%s/queue/id:%06u,symcc",
```

Comment:
I don't care about this so much, but this overriding forces all the names of seed files (including even the ones AFL generated) to be "id:****, symcc", right?
I think it would be clearer if the seed files SymCC generated and the ones AFL generated have the different file names. What do you think about this?

Reply:
oh I'm mistaken. Forget this please.